src/components/google-analytics.tsx

```
@@ -0,0 +1,109 @@
+import Script from "next/script";
+import { useEffect, useState } from "react";
+
+export function GoogleAnalytics() {
+  const gaId = process.env.NEXT_PUBLIC_GA_ID;
+  const [consentGiven, setConsentGiven] = useState(false);
+
+  if (!gaId) {
+    return null;
+  }
+
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)